include/mbedtls/asn1write.h

```diff
@@ -167,6 +182,21 @@ int mbedtls_asn1_write_int( unsigned char **p, unsigned char *start, int val );
 int mbedtls_asn1_write_printable_string( unsigned char **p, unsigned char *start,
                                  const char *text, size_t text_len );
 
+/**
+ * \brief           Write a UTF8 string tag (MBEDTLS_ASN1_UTF8_STRING) and
+ *                  value in ASN.1 format
+ *                  Note: function works backwards in data buffer
+ *
+ * \param p         reference to current position pointer
+ * \param start     start of the buffer (for bounds-checking)
+ * \param text      the text to write
+ * \param text_len  length of the text
+ *
+ * \return          the length written or a negative error code
```

Comment:
Pre-existing, but it should be clarified that "length" refers to the entire ASN.1 data, so `The number of bytes written to the target buffer \c *p on success.` Also, `a negative error code` should be in a separate line, i.e. `\return A negative error code on failure.`

@thomas-dee Do you mind making these changes while at it?